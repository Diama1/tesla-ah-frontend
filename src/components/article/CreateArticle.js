/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable jsx-a11y/img-redundant-alt */
/* eslint-disable jsx-a11y/alt-text */
/* eslint-disable jsx-a11y/anchor-is-valid */
/* eslint-disable react/no-unused-state */
import React, { Component } from 'react';
import Dante from 'Dante2';
import { Header } from '../layouts/Header';
/**
 * @description - a form which will handle the creation of an article
 */
class CreateArticle extends Component {
  state = {
    title: '',
    body: '',
    description: '',
  };

  handleOnChange = (editor) => {
    // eslint-disable-next-line
    console.log('editor content', editor.emitSerializedOutput());
  };

  render() {
    return (
      <div>
        <Header />
        <div className="bg-primary">
          <Dante onChange={this.handleOnChange} />
        </div>
      </div>
    );
  }
}

export default CreateArticle;
