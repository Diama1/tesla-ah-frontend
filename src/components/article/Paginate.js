/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-script-url */
/* eslint-disable react/prop-types */
/* eslint-disable no-plusplus */
/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { Fragment } from 'react';
import Pages from 'lodash';

const Paginate = ({
  totalArticles, pageSize, onPageChange,
}) => {
  const pagesCount = Math.ceil(totalArticles / pageSize);
  if (pagesCount === 0) return null;
  const pages = Pages.range(1, pagesCount + 1);
  return (
    <Fragment>
      <ul className="pagination">
        {pages.map(number => (
          <li key={number} className="page-item">
            <a onClick={() => onPageChange(number)} className="page-link">
              {number}
            </a>
          </li>
        ))}
      </ul>
    </Fragment>
  );
};

export default Paginate;
