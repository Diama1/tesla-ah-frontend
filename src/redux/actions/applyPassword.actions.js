import axios from 'axios';
import { PATCH_RESET } from './types/resetPassword.type';
import { setLoading, setLoaded } from './ui.actions';
import { BACKEND_URL } from '../../utils/constants';


/**
 * @returns {*} dispatch
 * @param {string} newpassword
 * @param {string} token
 */
const applyPassword = (newpassword, token) => async (dispatch) => {
  dispatch(setLoading());
  try {
    const res = await axios.patch(`${BACKEND_URL}/api/auth/reset/?token=${token}`, { newpassword });
    dispatch(setLoaded());
    dispatch({
      type: PATCH_RESET,
      payload: res.data,
    });
  } catch ({ response }) {
    dispatch(setLoaded());
    dispatch({
      type: PATCH_RESET,
      payload: response.data,
    });
  }
};
export default applyPassword;
