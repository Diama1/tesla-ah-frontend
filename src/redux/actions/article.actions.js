/* eslint-disable consistent-return */
import Axios from 'axios';
import {
  GET_ARTICLES,
  GET_ARTICLE,
  ARTICLE_ERRORS,
} from './types/article.type';
import { BACKEND_URL } from '../../utils/constants';

export const getArticles = () => (dispatch) => {
  Axios.get(`${BACKEND_URL}/api/articles`)
    .then((res) => {
      dispatch({
        type: GET_ARTICLES,
        payload: res.data,
      });
    })
    .catch((err) => {
      dispatch({
        type: ARTICLE_ERRORS,
        payload: err,
      });
    });
};

export const getOneArticle = slug => (dispatch) => {
  Axios.get(`${BACKEND_URL}/api/articles/${slug}`)
    .then((res) => {
      dispatch({
        type: GET_ARTICLE,
        payload: res.data.article,
      });
    })
    .catch((err) => {
      dispatch({
        type: ARTICLE_ERRORS,
        payload: err,
      });
    });
};
