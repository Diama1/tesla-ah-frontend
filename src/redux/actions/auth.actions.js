import axios from 'axios';
import { toast } from 'react-toastify';
import { setLoading, setLoaded } from './ui.actions';
import { BACKEND_URL } from '../../utils/constants';
import {
  VERIFIED,
  SIGNUP_SUCCESS,
  VERIFICATION_FAILED,
  LOG_OUT_SUCCESS,
  LOG_OUT,
} from './types/auth.type';

export const createAccount = userInfo => async (dispatch) => {
  dispatch(setLoading());
  try {
    // eslint-disable-next-line no-unused-vars
    const { data } = await axios.post(
      `${BACKEND_URL}/api/auth/signup`,
      userInfo,
    );
    dispatch(setLoaded());
    // eslint-disable-next-line no-undef
    sessionStorage.setItem('token', data.data.token);
    localStorage.user = JSON.stringify(data.data);
    toast.success('You are now registered to AH, check your email to verify your account');
    dispatch({
      type: SIGNUP_SUCCESS,
    });
  } catch (error) {
    dispatch(setLoaded());
    const { message } = error.response.data;
    toast.error(message);
  }
};

export const verifyAccount = token => async (dispatch) => {
  dispatch(setLoading());
  try {
    await axios.get(`${BACKEND_URL}/api/auth/verify/?token=${token}`);
    dispatch(setLoaded());
    dispatch({
      type: VERIFIED,
    });
  } catch (error) {
    toast.error('Invalid request');
    dispatch(setLoaded());
    dispatch({
      type: VERIFICATION_FAILED,
    });
  }
};

export const loggOut = () => async (dispatch) => {
  try {
    const { data } = await axios.get(`${BACKEND_URL}/api/auth/signout`, {
      headers: { token: sessionStorage.getItem('token') },
    });
    dispatch({
      type: LOG_OUT_SUCCESS,
      payload: { message: data.message },
    });
  } catch (error) {
    dispatch({
      type: LOG_OUT,
      payload: error,
    });
  }
};
