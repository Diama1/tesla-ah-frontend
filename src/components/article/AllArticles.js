/* eslint-disable no-unreachable */
/* eslint-disable no-else-return */
/* eslint-disable consistent-return */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/anchor-is-valid */
/* eslint-disable no-script-url */
/* eslint-disable jsx-a11y/anchor-has-content */
/* eslint-disable react/jsx-indent-props */
/* eslint-disable no-plusplus */
/* eslint-disable react/sort-comp */
/* eslint-disable no-console */
/* eslint-disable react/button-has-type */
/* eslint-disable jsx-a11y/img-redundant-alt */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/forbid-prop-types */
/* eslint-disable react/no-unused-prop-types */
/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import moment from 'moment';
import { toast } from 'react-toastify';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { getArticles } from '../../redux/actions/article.actions';
import { setLoading, setLoaded } from '../../redux/actions/ui.actions';
import Paginate from './Paginate';
import pagination from '../../utils/pagination';

export class AllArticles extends Component {
  state = {
    currentPage: 1,
    isClicked: false,
    openSlug: null,
  };

  componentDidMount() {
    this.props.getArticles();
  }

  handlePageChange = (pageNumber) => {
    this.setState({ currentPage: pageNumber });
  };

  handleToggle = (slug) => {
    const { isClicked } = this.state;
    this.setState({ isClicked: !isClicked, openSlug: slug });
  };

  render() {
    const {
      articles: { articles: { articles } },
    } = this.props;
    const { isClicked, currentPage, openSlug } = this.state;
    const pageSize = 5;
    const paginateArticles = pagination(articles, currentPage, pageSize);
    if (articles) {
      toast.success('Fetching Articles succeeded...');
      return (
        <div className="main-section container mt-5 mb-5">
          <br />
          <div className="row ml-2">
            <div className="col-lg-12">
              <label
                htmlFor="story_title"
                className="h3 font-weight-bold text-dark"
              >
                {' '}
                My Stories
              </label>
            </div>
          </div>
          {paginateArticles.map(article => (
            <div className="row mt-5 mb-3 ml-3 " key={article.id}>
              <div
                className="col-lg-8 border-bottom"
                style={{ cursor: 'pointer' }}
              >
                <Link
                  to={`/articles/${article.slug}`}
                  style={{ textDecoration: 'none' }}
                >
                  <div className="row">
                    <label
                      htmlFor="story_title"
                      className="h5 text-dark font-weight-bold"
                    >
                      {article.title}
                    </label>
                  </div>
                  <div className="row">
                    <label
                      htmlFor="story_description"
                      className="text-secondary font-weight-light"
                    >
                      {`${article.description.substring(1, 350)}...`}
                    </label>
                  </div>
                  <div className="row">
                    <div className="col-lg-0">
                      <label
                        htmlFor="story_date"
                        className="h6 text-secondary font-weight-bold"
                      >
                        {moment(article.createdAt).format('DD MMMM YYYY')}
                      </label>
                    </div>
                    <div className="col-lg-6">
                      <label
                        htmlFor="story_readtime"
                        className="h6 text-secondary font-weight-normal"
                      >
                        {article.readtime}
                      </label>
                    </div>
                    <div className="col-lg-8" />
                  </div>
                </Link>
              </div>
              <div className="col-lg-2">
                <img
                    src={JSON.parse(article.body).article.body.blocks[2].data.url}
                    alt="Story1-Image"
                    className="rounded-0 w-100"
                />
              </div>
              <div className="col-lg-1">
                <i
                  className="fas fa-ellipsis-v text-secondary icon-hover"
                  style={{ fontSize: '20px' }}
                  onClick={this.handleToggle.bind(this, article.slug)}
                />
                {isClicked && openSlug === article.slug ? (
                  <div className="bookmarks_button">
                    <button
                      className="text-dark w-100 border-0"
                    >
                      Remove
                    </button>
                    <button
                      className="text-dark w-100 border-0"
                    >
                      Edit
                    </button>
                  </div>
                ) : null}
              </div>
            </div>
          ))}
          <Paginate
            totalArticles={articles.length}
            pageSize={pageSize}
            currentPage={currentPage}
            onPageChange={this.handlePageChange}
          />
        </div>
      );
    } else {
      return (
        <div className=" container">
          <div className="row">
            <div className="col-lg-3" />
            <div className="col-lg-6 mt-5">
              <h3>
                {' '}
                { this.props.articles.articles.message }
                {' '}
              </h3>
            </div>
            <div className="col-lg-3" />
          </div>
        </div>
      );
    }
    return <h2> Loading... </h2>;
  }
}

const mapStateToProps = state => ({
  articles: state.article,
  ui: state.ui,
});

export default connect(
  mapStateToProps,
  { getArticles, setLoaded, setLoading },
)(AllArticles);
