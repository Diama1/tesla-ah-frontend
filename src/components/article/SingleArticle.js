/* eslint-disable no-console */
/* eslint-disable import/no-extraneous-dependencies */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getOneArticle } from '../../redux/actions/article.actions';

class SingleArticle extends Component {
  componentDidMount() {
    // eslint-disable-next-line react/prop-types
    const { match: { params: { slug } } } = this.props;
    // eslint-disable-next-line
    this.props.getOneArticle(slug);
  }

  render() {
    // eslint-disable-next-line
    const { article: { title, readtime, body } } = this.props;
    return (
      <div>
        <div className="row">
          <div className="col-lg-2" />
          <div className="col-lg-8">
            <div className="row">
              <div className="col-lg-12">
                {title}
              </div>
            </div>
            <div className="row">
              <div className="col-lg-12">
                {body}
              </div>
            </div>
            <div className="row">
              <div className="col-lg-12">
                {readtime}
              </div>
            </div>
          </div>
          <div className="col-lg-2" />
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  article: state.article,
});

export default connect(mapStateToProps, { getOneArticle })(SingleArticle);
