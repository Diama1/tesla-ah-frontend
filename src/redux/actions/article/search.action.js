import axios from 'axios';
// import { toast } from 'react-toastify';
import * as types from '../types/article.type';
import { BACKEND_URL } from '../../../utils/constants';

export default (queryText, type) => async (dispatch) => {
  dispatch({
    type: types.SEARCH_CLEAR,
  });
  dispatch({
    type: types.SEARCH_PENDING,
  });
  try {
    const { data: response } = await axios.get(`${BACKEND_URL}/api/articles/?${type}=${queryText}`);
    dispatch({
      type: types.SEARCH_DONE,
      payload: response.data,
    });
  } catch ({ response: { data } }) {
    dispatch({
      type: types.SEARCH_FAILED,
      payload: data.error,
    });
  }
};
