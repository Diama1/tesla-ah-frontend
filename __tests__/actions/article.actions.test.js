/* eslint-disable no-undef */
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { getArticles, getOneArticle } from '../../src/redux/actions/article.actions';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('FETCH ARTICLES action', () => {
  let store;
  beforeEach(() => {
    store = mockStore({});
  });
  it('Fetch GET_ARTICLES when the articles found', () => {
    store.dispatch(getArticles());
    expect(store.getActions()).toMatchSnapshot();
  });

  it('Fetch GET_ARTICLE when the article found', () => {
    store.dispatch(getOneArticle('this-text-is-copied-from-igihe-1ff4jyyfya'));
    expect(store.getActions()).toMatchSnapshot();
  });
});
