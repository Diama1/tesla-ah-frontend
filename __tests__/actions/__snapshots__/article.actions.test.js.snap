// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FETCH ARTICLES action Fetch GET_ARTICLE when the article found 1`] = `Array []`;

exports[`FETCH ARTICLES action Fetch GET_ARTICLES when the articles found 1`] = `Array []`;
